Packages/com.unity.inputsystem/InputSystem/Actions/InputActionAsset.cs

```diff
@@ -1014,60 +1014,125 @@ internal void MigrateJson(ref ReadFileJson parsedJson)
         {
             if (parsedJson.version >= JsonVersion.Version1)
                 return;
-            if ((parsedJson.maps?.Length ?? 0) > 0 && (parsedJson.version) < JsonVersion.Version1)
+
+            if (parsedJson.maps == null || parsedJson.maps.Length == 0)
+            {
+                parsedJson.version = JsonVersion.Version1;
+                return;
+            }
+
+            for (var mi = 0; mi < parsedJson.maps.Length; ++mi)
             {
-                for (var mi = 0; mi < parsedJson.maps.Length; ++mi)
+                var mapJson = parsedJson.maps[mi];
+                if (mapJson.actions == null || mapJson.actions.Length == 0)
+                    continue;
+
+                for (var ai = 0; ai < mapJson.actions.Length; ++ai)
                 {
-                    var mapJson = parsedJson.maps[mi];
-                    for (var ai = 0; ai < mapJson.actions.Length; ++ai)
+                    var actionJson = mapJson.actions[ai];
+                    var raw = actionJson.processors;
+                    if (string.IsNullOrEmpty(raw))
+                        continue;
+
+                    var parts = System.Text.RegularExpressions.Regex.Split(raw, @"\s*([,;])\s*");
+                    if (parts.Length == 0)
+                        continue;
+
+                    var tokens = new List<string>();
+                    for (int i = 0; i < parts.Length; i += 2)
+                        if (!string.IsNullOrEmpty(parts[i]))
+                            tokens.Add(parts[i]);
+
+                    if (tokens.Count == 0)
+                        continue;
+
+                    var parsed = new List<NameAndParameters>(tokens.Count);
```

Comment:
Also this list may be moved to outer scope to avoid allocating on every iteration.